File: src/main/java/itmo/is/project/mapper/module/ProductionModuleMapper.java
package itmo.is.project.mapper;

import itmo.is.project.dto.ProductionModuleDto;
import itmo.is.project.model.module.production.ProductionModule;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring", uses = {ProductionModuleBlueprintMapper.class})
public interface ProductionModuleMapper extends EntityMapper<ProductionModuleDto, ProductionModule> {

    @Override
    @Mapping(target = "productionModuleBlueprintDto", source = "blueprint")
    ProductionModuleDto toDto(ProductionModule entity);

    @Override
    @Mapping(target = "blueprint", source = "productionModuleBlueprintDto")
    ProductionModule toEntity(ProductionModuleDto dto);
}

File: src/main/kotlin/itmo/isproject/mapper/module/production/ProductionModuleMapper.kt
package itmo.isproject.mapper.module.production

import itmo.isproject.dto.module.production.ProductionModuleDto
import itmo.isproject.mapper.EntityMapper
import itmo.isproject.model.module.production.ProductionModule
import org.mapstruct.Mapper

@Mapper(uses = [ProductionModuleBlueprintMapper::class])
interface ProductionModuleMapper : EntityMapper<ProductionModuleDto, ProductionModule>
